src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -20,11 +20,47 @@ public class TemplateBinder
         private readonly RouteValueDictionary _filters;
         private readonly RouteTemplate _template;
 
+        /// <summary>
+        /// <para>
+        /// This constructor is obsolete and will be removed in a future version. The recommended
+        /// alternative is the overload that does not take a UrlEncoder.
+        /// </para>
+        /// <para>Creates a new instance of <see cref="TemplateBinder"/>.</para>
+        /// </summary>
+        /// <param name="urlEncoder">The <see cref="UrlEncoder"/>.</param>
+        /// <param name="pool">The <see cref="ObjectPool{T}"/>.</param>
+        /// <param name="template">The <see cref="RouteTemplate"/> to bind values to.</param>
+        /// <param name="defaults">The default values for <paramref name="template"/>.</param>
+        [Obsolete("This constructor is obsolete and will be removed in a future version. The recommended " +
+            "alternative is the overload that does not take a UrlEncoder.")]
         public TemplateBinder(
             UrlEncoder urlEncoder,
             ObjectPool<UriBuildingContext> pool,
             RouteTemplate template,
             RouteValueDictionary defaults)
+            : this(pool, template, defaults, urlEncoder)
```

Comment:
Just drop `urlEncoder` and call the new (3-parameter) constructor. If tests require `urlEncoder` to do something, have both constructors set `_urlEncoder`: This one should set it to `urlEncoder` and the new constructor should set it to `UrlEncoder.Default`.